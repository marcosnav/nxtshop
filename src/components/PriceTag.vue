<template>
  <div class="PriceTag">
    <slot />
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
export default Vue.extend({});
</script>

<style lang="sass" scoped>
.PriceTag
  background: #fff
  border-radius: 6px
  box-shadow: 0 3px 6px 0px rgba(0,0,0,0.16)
  display: inline-block
  font-size: 1.44em
  line-height: 42px
  margin-bottom: 10px
  padding: 0 16px
  position: relative
</style>
