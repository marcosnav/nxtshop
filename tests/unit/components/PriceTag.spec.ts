import { shallowMount } from '@vue/test-utils';
import PriceTag from '@/components/PriceTag.vue';

describe('<PriceTag>', () => {
  test('renders properly', () => {
    const component = shallowMount(PriceTag, {
      slots: {
        default: '$199',
      },
    });
    expect(component.classes('PriceTag')).toBe(true);
    expect(component.text()).toMatch('$199');
    expect(component.element).toMatchSnapshot();
  });
});
