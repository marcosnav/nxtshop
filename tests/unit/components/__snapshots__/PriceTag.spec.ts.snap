// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriceTag> renders properly 1`] = `
<div
  class="PriceTag"
>
  $199
</div>
`;
